app/actions/remote/entry/common.ts

```diff
@@ -60,7 +63,65 @@ export const teamsToRemove = async (serverUrl: string, removeTeamIds?: string[])
     return undefined;
 };
 
-export const fetchAppEntryData = async (serverUrl: string, since: number, initialTeamId: string): Promise<AppEntryData | AppEntryError> => {
+export type EntryResponse = {
+    models: Model[];
+    initialTeamId: string;
+    initialChannelId: string;
+    prefData: MyPreferencesRequest;
+    teamData: MyTeamsRequest;
+    chData?: MyChannelsRequest;
+} | {
+    error: unknown;
+}
+export const entry = async (serverUrl: string, teamId?: string, channelId?: string, since = 0): Promise<EntryResponse> => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const {database} = operator;
+
+    const lastDisconnectedAt = since || await getWebSocketLastDisconnected(database);
+
+    const fetchedData = await fetchAppEntryData(serverUrl, lastDisconnectedAt, teamId);
+    if ('error' in fetchedData) {
+        return {error: fetchedData.error};
+    }
+
+    const {initialTeamId, teamData, chData, prefData, meData, removeTeamIds, removeChannelIds} = fetchedData;
+    const error = teamData.error || chData?.error || prefData.error || meData.error;
+    if (error) {
+        return {error};
```

Comment:
This may be returning too soon (we used to do this check the errors at the end) but I think it is better be safe than sorry (bad state in the database). 0/5